File: contest_of_dart/test/challenges/challenge2_test.dart
import 'package:test/test.dart';
import '../../bin/challenges/challenge2.dart';

void main() {
  test('is anagram of', () {
    expect(
      isWord($: "listen", anagramOf: "silent"),
      equals(true)
    );
    expect(
        isWord($: "triangle", anagramOf: "integral"),
        equals(true)
    );
    expect(
        isWord($: "amor", anagramOf: "roma"),
        equals(true)
    );
    expect(
        isWord($: "sergio", anagramOf: "riesgo"),
        equals(true)
    );
    expect(
        isWord($: "delira", anagramOf: "lidera"),
        equals(true)
    );
    expect(
        isWord($: "ballena", anagramOf: "llenaba"),
        equals(true)
    );
    expect(
        isWord($: "valora", anagramOf: "Alvaro"),
        equals(true)
    );
    expect(
        isWord($: "PRAGA", anagramOf: "pagar"),
        equals(true)
    );

    expect(
        isWord($: "amor", anagramOf: "amor"),
        equals(false)
    );
    expect(
        isWord($: "edwin", anagramOf: "winned"),
        equals(false)
    );
    expect(
        isWord($: "obito", anagramOf: "tobi"),
        equals(false)
    );
    expect(
        isWord($: "sasuke", anagramOf: "itachi"),
        equals(false)
    );
    expect(
        isWord($: "maria", anagramOf: "rima"),
        equals(false)
    );
    expect(
        isWord($: "alborota", anagramOf: "lobo rata"),
        equals(false)
    );
  });
}

File: contestOfKotlin/src/test/kotlin/challenges/Challenge2Test.kt
package challenges.challenge2

import org.example.challenges.challenge2.*
import kotlin.test.*

class Challenge2TestCase {
    @Test fun testIsWordAnagramOf() {
        assertTrue(
            isWord("listen", anagramOf = "silent")
        )
        assertTrue(
            isWord("triangle", anagramOf = "integral")
        )
        assertTrue(
            isWord("amor", anagramOf = "roma")
        )
        assertTrue(
            isWord("sergio", anagramOf = "riesgo")
        )
        assertTrue(
            isWord("delira", anagramOf = "lidera")
        )
        assertTrue(
            isWord("ballena", anagramOf = "llenaba")
        )
        assertTrue(
            isWord("valora", anagramOf = "Alvaro")
        )
        assertTrue(
            isWord("PRAGA", anagramOf = "pagar")
        )

        assertFalse(
            isWord("amor", anagramOf = "amor")
        )
        assertFalse(
            isWord("edwin", anagramOf = "winned")
        )
        assertFalse(
            isWord("obito", anagramOf = "tobi")
        )
        assertFalse(
            isWord("sasuke", anagramOf = "itachi")
        )
        assertFalse(
            isWord("maria", anagramOf = "rima")
        )
        assertFalse(
            isWord("alborota", anagramOf = "lobo rata")
        )
    }
}